src/ifcsverchok/__init__.py

```diff
@@ -84,7 +96,131 @@ def make_node_list():
 reload_event = False
 
 import bpy
+import os
+from os.path import abspath, splitext
+import ifcopenshell
+from ifcsverchok.ifcstore import SvIfcStore
+from sverchok.data_structure import flatten_data
+
+class IFC_Sv_UpdateCurrent(bpy.types.Operator):
+    """Update current Sverchok node tree"""
+    bl_idname = "ifc.sverchok_update_current"
+    bl_label = "Update current node tree"
+    bl_options = {'REGISTER', 'UNDO', 'INTERNAL'}
+
+    node_group: bpy.props.StringProperty(default="")
+    force_mode: bpy.props.BoolProperty(default=False)
+
+    def execute(self, context):
+        print("#"*10, "Update current node tree")
+        self.file = SvIfcStore.purge()
+        self.file = SvIfcStore.get_file()
+        self.file.write("/Users/martina/Documents/GSoC/CodeTests/IfcFileTest_7_11_purged.ifc")
+        node_tree = context.space_data.node_tree
+        if node_tree:
+            if self.force_mode or node_tree.sv_process:
+                try:
+                    bpy.context.window.cursor_set("WAIT")
+                    node_tree.force_update()
+                finally:
+                    bpy.context.window.cursor_set("DEFAULT")
+        self.report({"INFO"}, "Node tree updated.")
+        return {'FINISHED'}
+
+class IFC_Sv_write_file(bpy.types.Operator):
+    bl_idname = "ifc.write_file_panel"
+    bl_label = "Write File"
+    bl_options = {"REGISTER", "UNDO"}
+    bl_description = "File path to write to."
+    filepath: bpy.props.StringProperty(subtype="FILE_PATH")
+    node_group: bpy.props.StringProperty(default="")
+    force_mode: bpy.props.BoolProperty(default=False)
+
+    @classmethod
+    def poll(cls, context):
+        return any("IFC" in n for n in context.space_data.edit_tree.nodes.keys())
+
+    def ensure_hirarchy(self, file):
+        elements_in_buildings = []
+        if not 0 <= 0 < len(file.by_type("IfcBuilding")):
```

Comment:
This is an interesting if statement. Is that `if len(file.by_type("IfcBuilding")) == 0`?

Reply:
in this case yes. It was a way to check if index exists without throwing an IndexError